### ISSUE TITLE
Could you provide the training(overfitting) script to generate the weight and bias of SIREN?

### ISSUE THREAD

**Issue by dongzhuoyao** [AvivNavon/DWSNets#6]

**Comment by AvivNavon** [comment]
Hi @dongzhuoyao ,
The code needs some refactoring before we can publish it.
For now, let me give some details so you can reproduce our data generation procedure:
- We use the INR class [here](https://github.com/AvivNavon/DWSNets/blob/main/nn/inr.py#L61) with `up_scale=16`.
- We use Adam optimizer with lr 5e − 4 (for MNIST / Fashion-MNIST).
- We use MSE loss and train the INR for 1K steps with early stopping if we reach 40 PSNR.

**Comment by AvivSham** [comment]
Closed due to inactivity.

**Comment by canerozer** [comment]
Hello,

I have attempted to train DWSNet after trying to reproduce the weight spaces corresponding to the MNIST INRs. However, I am not able to achieve similar representations and results. Could you share the INR generation code and the running instructions? With my reproductions, I can barely reach 65% testing accuracy.

For reference, I have included the UMAP projections of the weight spaces of your shared INRs and my reproductions below:

![image (1)](https://github.com/user-attachments/assets/096f6e21-4a7b-4eb3-8205-23456b73e2cd)
![image](https://github.com/user-attachments/assets/ff14cb2b-3c14-4445-b62d-84145ce68b4d)

Thanks!
